configuration/config2/SchemaValue.cpp

```diff
@@ -204,69 +204,68 @@ void SchemaValue::getAllowedValues(std::vector<AllowedValue> &allowedValues, con
     }
 
     //
-    // Is there a specialized rule that limits the values?
-    else if (!m_valueLimitRuleType.empty())
+    // Is there a specialized rule that limits the values? The order is important here. Place more restrictive rules first. Note also that
+    // a rule could be used inside a unique value set which should remain the last entry in this if then else if block
+
+    //
+    // uniqueItemType_1 - value is based on a unique item type described by the data for the rule. This is version 1
```

Comment:
based on the name, we might have multiple uniqueitemtypes, let's consider representing as enumeration if it makes sense. Also, let's consider giving them more self documenting names. For example uniqueItemType_Counter or uniqueItemType_RandomAlphChars, etc.

Reply:
I made the names of the rule types more descriptive. They are also documented in the Wiki. For now I will leave the reference to strings as they match what is found in the XSD and should someone want to find application of such rules, searching is easier.